manual.typ

```diff
@@ -1415,3 +1374,187 @@ tree.tree(data, content: (padding: .1), direction: "right",
+    plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+    plot.add(domain: (0, 360), x => calc.cos(x * 1deg),
+             samples: 10, mark: "x", mark-style: (stroke: blue))
+  })
+}, ```typc
+plot.plot(size: (2,2), x-tick-step: 180, y-tick-step: 1,
+          x-unit: $degree$, y-max: .5, {
+  plot.add(domain: (0, 360), x => calc.sin(x * 1deg))
+  plot.add(domain: (0, 360), x => calc.cos(x * 1deg),
+           samples: 10, mark: "x", mark-style: (stroke: blue))
+})
+```)
+
+== Chart
+
+With the `chart` library it is easy to draw charts.
+
+=== Barchart
+
+```typc
+barchart(size: (width, height))
+```
+
+#def-arg("data", "a",
+  [Data array of arrays or dictionaries])
+#def-arg("label-key", "s|i",
+  [Data item key to access an items label])
+#def-arg("value-key", "s|i|a",
+  [Data item key(s) to access an items value(s). For multi-value charts
+   this must be an array of all keys, e.G. `(..range(1, 5))`])
+#def-arg("mode", "s",
+  [Barchart mode, one of "basic", "clustered" (bars next to each other), "stacked" (bar stacked) or "stacked100" (bars stacked but as percentage of their sum)])
+#def-arg("size", "a",
+  [The chart's size. Height can be set to `auto`.])
+#def-arg("bar-width", "f",
+  [Width of a bar or cluster of bars, with $1$ being leving no gap between
+   values.])
+#def-arg("bar-style", "function|style",
+  [Style of bars, accepts a function of the form `index => style`. You canvas    use palettes from the `palette` library, see @palette.])
+#def-arg("x-tick-step", "f",
+  [X axis tick step])
+#def-arg("x-ticks", "a",
+  [X axis ticks list, a list of tick values or value/label tuples])
+#def-arg("x-unit", "content",
+  [X axis tick label suffix])
+#def-arg("x-label", "content?",
+  [X axis label])
+#def-arg("y-label", "content?",
+  [Y axis label])
+
+=== Examples
+==== Basic
+#example({
+  let data = (("A", 10), ("B", 20), ("C", 13))
+  chart.barchart(size: (3, auto), x-tick-step: 10, data)
+}, ```typc
+let data = (("A", 10), ("B", 20), ("C", 13))
+chart.barchart(size: (3, auto), x-tick-step: 10, data)
+```)
+
+==== Clustered
+#example({
+  let data = (("A", 10, 12, 22), ("B", 20, 1, 7), ("C", 13, 8, 9))
+  chart.barchart(size: (3, auto), mode: "clustered",
+                 x-tick-step: 10, value-key: (..range(1, 4)), data)
+}, ```typc
+let data = (("A", 10, 12, 22), ("B", 20, 1, 7), ("C", 13, 8, 9))
+chart.barchart(size: (3, auto), mode: "clustered",
+               x-tick-step: 10, value-key: (..range(1, 4)), data)
+```)
+
+==== Stacked
+#example({
+  let data = (("A", 10, 12, 22), ("B", 20, 1, 7), ("C", 13, 8, 9))
+  chart.barchart(size: (3, auto), mode: "stacked",
+                 x-tick-step: 10, value-key: (..range(1, 4)), data)
+}, ```typc
+let data = (("A", 10, 12, 22), ("B", 20, 1, 7), ("C", 13, 8, 9))
+chart.barchart(size: (3, auto), mode: "clustered",
+               x-tick-step: 10, value-key: (..range(1, 4)), data)
+```)
+
+== Palette <palette>
+
+A palette is a function that returns a style for an index.
```

Comment:
This is a pretty cool addition, an explantion of how to create one with an example wouldn't go amiss